src/rgw/rgw_sal.h

```diff
@@ -73,6 +73,9 @@ class RGWStore : public DoutPrefixProvider {
 			    req_info& req_info,
 			    std::unique_ptr<RGWBucket>* bucket) = 0;
     virtual RGWBucketList* list_buckets(void) = 0;
+    virtual bool is_meta_master() = 0;
```

Comment:
can be const

Reply:
What can be const?  Not the function, it uses svc(), which is part of the object itself.

Reply:
it sounds like we need a const overload of `svc()` too?

Reply:
I always have to look up the requirements for const methods, so maybe I'm missing something here.  I don't see how we can have a const svc(), it's inside rados, which is a non-const member, and can't (as far as I can see) be const?

Reply:
in general, const methods just have a `this` pointer of type `const T*`, so their members are also const. member pointers act a bit funny though, because it's the pointer itself that becomes immutable, not the pointed-to object.. i.e. you can mutate the pointed-to object, but can't assign a different memory location to the pointer

so in a const overload of svc(), the type of member `RGWRados *rados` becomes `RGWRados *const` rather than `const RGWRados*`. you _could_ return `&rados->svc` as a non-const `RGWServices*` pointer, but to preserve const-correctness you'd return it as `const RGWServices*`

```diff
     RGWServices *svc() { return &rados->svc; }
+    const RGWServices *svc() const { return &rados->svc; }
     RGWCtl *ctl() { return &rados->ctl; }
+    const RGWCtl *ctl() const { return &rados->ctl; }
```